modules/button/react/lib/IconButton.tsx

```diff
@@ -65,11 +70,31 @@ function getAccentSelector(fillColor: string): CSSObject {
   };
 }
 
+const changeStateSeparator = (shouldChange = false, obj?: CSSObject): CSSObject | undefined => {
+  if (!shouldChange || !obj) {
+    return obj;
+  }
+
+  return Object.keys(obj).reduce((result, key) => {
+    const newKey = key
+      .replace(/^:/, '&:')
+      .replace(/:/g, '.')
+      .replace(/\.not/g, ':not')
+      .replace(/\.disabled/, ':disabled');
+    const value =
+      typeof obj[key] === 'object' ? changeStateSeparator(true, obj[key] as CSSObject) : obj[key];
+    const newObj = {...result, [newKey]: value};
+    return newObj;
+  }, {});
+};
+
```

Comment:
@NicholasBoll and I wrote to this to be able to see the different states of IconButton. I think eventually we might want to move this into a utils?

Reply:
@mannycarrera4 @NicholasBoll what do you think about adding it to the [wrapper `styled` function](https://github.com/Workday/canvas-kit/blob/master/modules/_labs/core/react/lib/theming/styled.ts) we added for theming/RTL? You could check for the `changeStateSeparator` prop there, rather than having to add the complexity of calling it for every interpolation function in the component files.

Reply:
This is awesome btw 👌